src/WooCommerce.php

```diff
@@ -282,12 +293,15 @@ public static function isAttribute() {
    * @return string
    *   List of product brands.
    */
-  public static function getProductBrand($product_id) {
+  public static function getProductBrand($product_id, $parent_id) {
     $args = [
       'orderby' => 'name',
       'fields' => 'names',
     ];
     $brands = wp_get_post_terms($product_id, static::getProductBrandTermSlug(), $args);
+    if (empty($brands)) {
+      $brands = wp_get_post_terms($parent_id, static::getProductBrandTermSlug(), $args);
+    }
```

Comment:
If I'm not mistaken, a variant always gets its brand from the parent (please verify and confirm this yourself)
```suggestion
    $brands = wp_get_post_terms($parent_id ?: $product_id, static::getProductBrandTermSlug(), $args);
```

Reply:
That would had been great. Unfortunately not always the case.